src/Core/Content/Product/SalesChannel/Listing/Processor/SortingListingProcessor.php

```diff
@@ -91,34 +90,25 @@ private function getCurrentSorting(ProductSortingCollection $sortings, Request $
 
     private function getAvailableSortings(Request $request, Context $context): ProductSortingCollection
     {
-        $criteria = new Criteria();
-        $criteria->setTitle('product-listing::load-sortings');
-        /** @var string[] $availableSortings */
-        $availableSortings = $request->get('availableSortings');
-        $availableSortingsById = [];
+        /** @var ProductSortingCollection $availableSortings */
```

Comment:
see my other comment. this is not true, as you have an array of product sortings